### ISSUE TITLE
[Bug]: Notifications.hasPermission() does not always return a boolean on iOS

### ISSUE THREAD

**Issue by Bjarkinn** [OneSignal/react-native-onesignal#1555]
### What happened?

If the notification permission is checked after the permission is manually changed by the user in the settings the `Notifications.hasPermission()` function returns the following object `{permission: boolean}` instead of boolean

#### Video

https://github.com/OneSignal/react-native-onesignal/assets/24234405/5c683a4a-56ad-46e2-b739-cc764b7ba974

In the video The following code was used that calls `Notifications.hasPermission()` every time the app comes into foreground
```js
const OneSignalIosBug: FC = () => {
  const [hasNotificationPermission, setHasNotificationPermission] = useState(
    OneSignal.Notifications.hasPermission()
  )

  useAppStateListener({
    onForeground: () => {
      const currNotificationPermission = OneSignal.Notifications.hasPermission()
      setHasNotificationPermission(currNotificationPermission)
    },
  })

  return (
    <Text style={styles.oneSignalPermissionText}>
      {(isAndroid ? 'Android: ' : 'iOS: ') +
        JSON.stringify(hasNotificationPermission)}
    </Text>
  )
}

const styles = StyleSheet.create({
  oneSignalPermissionText: {
    ...StyleSheet.absoluteFillObject,
    textAlign: 'center',
    top: '50%',
  },
})

export default React.memo(OneSignalIosBug)

```

### Steps to reproduce?

```Markdown
1. Installed packages

"react": "18.2.0",
"react-native": "0.72.4",
"expo": "49.0.9",
"react-native-onesignal": "5.0.0",

2. Check the permission with `OneSignal.Notifications.hasPermission()` that returns a boolean
3. Go to the iOS app setting and change the notification permission manually
4. Check the permission again with `OneSignal.Notifications.hasPermission()` that returns an object {permission: boolean}
```

### What did you expect to happen?

I expected `Notifications.hasPermission()` to always return a boolean

### React Native OneSignal SDK version

5.0.0

### Which platform(s) are affected?

- [X] iOS
- [ ] Android

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jennantilla** [comment]
Hello @Bjarkinn thank you for flagging! We'll work on a fix for this and let you know when it's released!

**Comment by ChromeQ** [comment]
On a related note there is a bug reported here also related to `hasPermission` https://github.com/OneSignal/react-native-onesignal/issues/1506#issuecomment-1706332448

Perhaps these can be investigated together. fyi this one also affects android not just ios

**Comment by ChromeQ** [comment]
@Bjarkinn @jennantilla I can also confirm this is also happening on Android, not just iOS. And not just the `hasPermission` method, I'm using the permissionChange event listener:

```ts
const handler = (granted: boolean) => {
  console.log('PERM CHANGE HANDLER', granted); // PERM CHANGE HANDLER {"permission": true}
};

OneSignal.Notifications.addEventListener('permissionChange', handler);
```

This is fine as I can extract the permission, maybe it is just the types need updating:
![image](https://github.com/OneSignal/react-native-onesignal/assets/625203/be3ceaf7-f07c-44bf-8e7f-e57a16f492e8)

**Comment by theonetheycallneo** [comment]
We did this for now:

```
      const hasPermission = await OneSignal.Notifications.requestPermission(true)
      //TODO: hasPermission is actually an object {permission: boolean}, bug on one signal.
      //@ts-ignore
      const isPermissionEnabled = hasPermission?.permission ?? hasPermission ?? false
      store.permissions.setIsPushDisabled(!isPermissionEnabled)
```

**Comment by jennantilla** [comment]
Hello everyone! Thank you for your patience. 

A fix for this is available in our latest release, [5.0.1](https://github.com/OneSignal/react-native-onesignal/releases/tag/5.0.1). Please let us know if you have any additional questions or concerns!

**Comment by Bjarkinn** [comment]
Looks like everything behaves like it should on 5.0.1
Good job 🤝

**Comment by Salmankhan033** [comment]
Same issue on V5.0.4 @jennantilla

**Comment by rakshitbharat** [comment]
This issue should be reopened because there is a new issue occurred in which the Boolean itself started returning false always.

**Comment by ChromeQ** [comment]
No this issue should not be reopened as you say:

> there is a new issue occurred 

It is a new issue, so please open a new issue instead of commenting on old ones.
The fix from this issue was released as [v5.0.1](https://github.com/OneSignal/react-native-onesignal/releases/tag/5.0.1) in [this PR](https://github.com/OneSignal/react-native-onesignal/pull/1558/files) - the relevant change there in `src/events/EventManager.ts` has not changed since v5.0.1 through to latest v5.0.5.
